File: 2.POO/Resto/Calificacion/1°version/Java/Calificacion.java
import java.util.Scanner;
public class Calificacion {
  public static void main(String[] args) {
    Scanner mex= new Scanner(System.in);

    Calificaciones C1 = new Calificaciones();
    System.out.println("\n \n");
    System.out.println("Ingresa tu nombre: ");
    C1.nombre= mex.nextLine();
    System.out.println("Ingresa el nombre de la materia: ");
    C1.materia= mex.nextLine();
    System.out.println("Ingresa la calificacion del cognitivo: ");
    C1.cognitivo= mex.nextDouble();
    System.out.println("Ingresa la calificacion del formativo: ");
    C1.formativo= mex.nextDouble();
    System.out.println("Ingresa la calificacion del sumativo: ");
    C1.sumativo= mex.nextDouble();
    System.out.println(C1.nombre + " su calificacion de " + C1.materia + " es: " + 
    ((C1.cognitivo * 0.5) + (C1.formativo * 0.25) + (C1.sumativo * 0.25)) + "\n.");

    System.out.println("\n");
    System.out.println("---------------------------------------------------------------------------");
    System.out.println("Creador: Jesus A. Mex Damian 20/02/2023.");


  }  
}

File: 2.POO/Resto/Calificacion/1°version/C++/Calificacion.cpp
#include <iostream>
#include <cstdlib>
using namespace std;
#include "Calificacion.h"

int main() {
    Calificacion C1;

    system("Color 03");
    cout << "\n \n";
    cout << "Ingresa tu nombre: ";
    getline(cin,C1.nombre);
    cout << "Nombre de la materia: ";
    getline(cin,C1.materia);
    cout << "Ingresa la calificacion del cognitivo: ";
    cin >> C1.cognitivo;
    if(C1.cognitivo >= 0 && C1.cognitivo <= 10){
        cout << "Ingresa la calificacion del sumativo: ";
        cin >> C1.sumativo;
        if(C1.sumativo <= 10 && C1.sumativo >= 0){
            cout << "Ingresa la calificacion del formativo: ";
            cin >> C1.formativo;
            if(C1.formativo <= 10 && C1.formativo >= 0){
                cout << C1.nombre << " su calificaion de " << C1.materia << " es: " << (C1.cognitivo * 0.5) + (C1.sumativo * 0.25) + (C1.formativo * 0.25);
            
            } else {cout << "Calificacion invalida";}


        } else {cout << "Calificacion invalida";}


    } else {cout << "Calificacion invalida";}

    cout << "\n \n";
    cout << "------------------------------------------------------------------------------------------\n";
	cout << "Creador: Jesus A. Mex Damian 15/02/2023.";
   
    return 0;
}
